### PR TITLE
Fixed doxygen warnings and make the build fail for undocumented code

### LINKED ISSUE #46: Fix Doxygen warnings
Thanks to the efforts of @richterjakob and @menon-karthik, we have very little undocumented code.

If we fix the last 60, we could make the Doxygen build fail on warnings to enforce that only documented new code can be merged. We would have to add `WARN_AS_ERROR = FAIL_ON_WARNINGS` to the Doxyfile (run through the whole build but throw an error if there's a warning).

The warnings are (mostly undocumented class variables):
```
Parsing file /Users/pfaller/work/r/Users/pfaller/work/repos/svZeroDPlus_fork/src/interface/interface.cpp:149: warning: documented symbol 'throw std::runtime_error' was not declared or defined.
src/interface/interface.cpp:153: warning: documented symbol 'throw std::runtime_error' was not declared or defined.
Building full member lists recursivel/Users/pfaller/work/repos/svZeroDPlus_fork/src/interface/interface.h:47: warning: Member S (typedef) of file interface.h is not documented.
src/model/ClosedLoopCoronaryBC.h:43: warning: Member Side (enumeration) of namespace zd_model is not documented.
src/model/ClosedLoopHeartPulmonary.h:41: warning: Member PI (macro definition) of file ClosedLoopHeartPulmonary.h is not documented.
src/optimize/calibrate.cpp:37: warning: Member calibrate(const nlohmann::json &config) (function) of namespace optimize is not documented.
src/solve/csv_writer.cpp:169: warning: The following parameter of io::to_variable_csv(std::vector< double > &times, std::vector< algebra::State > &states, zd_model::Model &model, bool mean, bool derivative) is not documented:
src/solve/csv_writer.cpp:38: warning: The following parameter of io::to_vessel_csv(std::vector< double > &times, std::vector< algebra::State > &states, zd_model::Model &model, bool mean, bool derivative) is not documented:
src/solve/debug.h:40: warning: Member DEBUG_MSG(str) (macro definition) of file debug.h is not documented.
src/interface/interface.h:55: warning: Member SolverInterface(const std::string &input_file_name) (function) of class SolverInterface is not documented.
src/interface/interface.h:61: warning: Member problem_id_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:62: warning: Member input_file_name_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:66: warning: Member external_step_size_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:69: warning: Member time_step_size_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:70: warning: Member num_time_steps_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:71: warning: Member absolute_tolerance_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:72: warning: Member max_nliter_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:73: warning: Member time_step_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:74: warning: Member save_interval_counter_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:75: warning: Member output_interval_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:76: warning: Member system_size_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:77: warning: Member num_output_steps_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:78: warning: Member pts_per_cycle_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:79: warning: Member output_last_cycle_only_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:81: warning: Member model_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:82: warning: Member integrator_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:84: warning: Member state_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:85: warning: Member times_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:86: warning: Member states_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:58: warning: Member problem_id_count_ (variable) of class SolverInterface is not documented.
src/interface/interface.h:59: warning: Member interface_list_ (variable) of class SolverInterface is not documented.
Generating docs for compound optimize::LevenbergMarquardtOptimizer.../Users/pfaller/work/repos/svZeroDPlus_fork/src/model/Block.h:225: warning: return type of member zd_model::Block::get_num_triplets is not documented
src/model/Block.h:131: warning: argument 'num_internal_vars' of command @param is not found in the argument list of zd_model::Block::setup_dofs_(DOFHandler &dofhandler, unsigned int num_equations, std::list< std::string > internal_var_names)
src/model/Block.h:131: warning: The following parameter of zd_model::Block::setup_dofs_(DOFHandler &dofhandler, unsigned int num_equations, std::list< std::string > internal_var_names) is not documented:
src/model/BloodVessel.h:147: warning: Member BloodVessel(int id, const std::vector< int > &param_ids, Model *model) (function) of class zd_model::BloodVessel is not documented.
src/model/BloodVessel.h:103: warning: unable to resolve reference to 'Parameters::stenosis_coefficient' for \ref command
src/model/BloodVessel.h:216: warning: return type of member zd_model::BloodVessel::get_num_triplets is not documented
src/model/BloodVesselJunction.h:214: warning: return type of member zd_model::BloodVesselJunction::get_num_triplets is not documented
src/model/ClosedLoopCoronaryBC.h:123: warning: Member ClosedLoopCoronaryBC(int id, const std::vector< int > &param_ids, Model *model, Side side) (function) of class zd_model::ClosedLoopCoronaryBC is not documented.
src/model/ClosedLoopCoronaryBC.h:189: warning: return type of member zd_model::ClosedLoopCoronaryBC::get_num_triplets is not documented
src/model/ClosedLoopHeartPulmonary.h:191: warning: return type of member zd_model::ClosedLoopHeartPulmonary::get_num_triplets is not documented
src/model/ClosedLoopRCRBC.h:165: warning: return type of member zd_model::ClosedLoopRCRBC::get_num_triplets is not documented
src/model/FlowReferenceBC.h:134: warning: return type of member zd_model::FlowReferenceBC::get_num_triplets is not documented
src/model/Junction.h:149: warning: Member num_inlets (variable) of class zd_model::Junction is not documented.
src/model/Junction.h:150: warning: Member num_outlets (variable) of class zd_model::Junction is not documented.
src/model/Junction.h:144: warning: return type of member zd_model::Junction::get_num_triplets is not documented
src/model/Model.h:137: warning: The following parameter of zd_model::Model::add_node(const std::vector< Block * > &inlet_eles, const std::vector< Block * > &outlet_eles, std::string_view name) is not documented:
src/model/Model.h:121: warning: argument 'block_id' of command @param is not found in the argument list of zd_model::Model::get_block_type(std::string_view name)
src/model/Model.h:121: warning: The following parameter of zd_model::Model::get_block_type(std::string_view name) is not documented:
src/model/Model.h:255: warning: parameters of member zd_model::Model::get_num_blocks are not documented
src/model/Node.h:66: warning: Member Node(int id, const std::vector< Block * > &inlet_eles, const std::vector< Block * > &outlet_eles, Model *model) (function) of class zd_model::Node is not documented.
src/model/Node.h:60: warning: zd_model::Node::Node has @param documentation sections but no arguments
src/model/Node.h:93: warning: unable to resolve reference to 'global_var_ids' for \ref command
src/model/Node.h:93: warning: unable to resolve reference to 'global_eqn_ids' for \ref command
src/model/OpenLoopCoronaryBC.h:160: warning: return type of member zd_model::OpenLoopCoronaryBC::get_num_triplets is not documented
src/model/Parameter.h:66: warning: The following parameter of zd_model::Parameter::Parameter(int id, const std::vector< double > &times, const std::vector< double > &values, bool periodic=true) is not documented:
src/model/PressureReferenceBC.h:135: warning: return type of member zd_model::PressureReferenceBC::get_num_triplets is not documented
src/model/ResistanceBC.h:133: warning: return type of member zd_model::ResistanceBC::get_num_triplets is not documented
src/model/ResistiveJunction.h:145: warning: return type of member zd_model::ResistiveJunction::get_num_triplets is not documented
src/model/WindkesselBC.h:157: warning: return type of member zd_model::WindkesselBC::get_num_triplets is not documented
```

### PR DESCRIPTION
**Pull request by menon-karthik**
Closes #46 

I fixed the previous warnings by adding the necessary documentation. The github actions should now fail and prevent undocumented code from being merged.

### CODE REVIEW

**Comment by menon-karthik** [comment]
The documentation is failing to build because the github actions can't find a package that it could find a few hours ago! I checked that it isn't from my changes by pushing a "test" commit [in this branch](https://github.com/StanfordCBCL/svZeroDPlus/compare/master...menon-karthik:svZeroDPlus:test_docs) where the only change from`StanfordCBCL:master` is that I removed one random word in the documentation in `Block.h`. So although that branch is essentially the same as `StanfordCBCL:master`, the documentation fails to build with the same error. @mrp089 @ktbolt have you seen this before?

**Comment by ktbolt** [comment]
@menon-karthik I've not seen this before, just tried to rerun it and it failed.

**Comment by menon-karthik** [comment]
@mrp089 did some digging and it turns out the package that is failing was updated today. Fingers crossed that this resolves itself.

**Comment by mrp089** [comment]
@menon-karthik, if you push to this branch now, does the documentation build?

**Comment by menon-karthik** [comment]
> @menon-karthik, if you push to this branch now, does the documentation build?

Nope.

**Comment by mrp089** [comment]
Try downgrading `mattnotmitt/doxygen-action`: https://github.com/mattnotmitt/doxygen-action

Maybe this only affects the `edge` version.

**Comment by mrp089** [comment]
@menon-karthik, happy to merge this first and then I'll deal with the conflicts in #74

File: src/interface/interface.cpp [hunk 1]
```diff
@@ -163,7 +162,7 @@ void initialize(std::string input_file_arg, int& problem_id, int& pts_per_cycle,
   interface->absolute_tolerance_ = simparams.sim_abs_tol;
   interface->time_step_ = 0;
   interface->system_size_ = model->dofhandler.size();
-  interface->output_interval_ = simparams.output_interval;
+  // interface->output_interval_ = simparams.output_interval;
```
Comment (mrp089) [inline src/interface/interface.cpp:165]: Not used? Remove?

**Review by mrp089 — CHANGES_REQUESTED** [review]
Looks great! Thanks for doing this! Just some minor things

**Review by mrp089 — APPROVED** [review]

**Pull request merged**